Jet soak test kinesis-test fails on 5.1-BETA-1 [HZ-815]

`kinesis-test` fails on `5.1-BETA-1` immediately after start with:
```
com.hazelcast.nio.serialization.HazelcastSerializationException: com.hazelcast.jet.JetException: java.io.IOException: unexpected exception type
        at com.hazelcast.internal.serialization.impl.SerializationUtil.handleException(SerializationUtil.java:111)
        at com.hazelcast.internal.serialization.impl.AbstractSerializationService.toObject(AbstractSerializationService.java:276)
        at com.hazelcast.jet.impl.execution.init.CustomClassLoadedObject.deserializeWithCustomClassLoader(CustomClassLoadedObject.java:66)
        at com.hazelcast.jet.impl.JobCoordinationService.deserializeJobDefinition(JobCoordinationService.java:1063)
        at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitJob$1(JobCoordinationService.java:238)
        at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$52(JobCoordinationService.java:1234)
        at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$53(JobCoordinationService.java:1255)
        at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64)
        at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
        at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
        at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)
Caused by: com.hazelcast.jet.JetException: java.io.IOException: unexpected exception type
        at com.hazelcast.jet.impl.util.ExceptionUtil.peeledAndUnchecked(ExceptionUtil.java:134)
        at com.hazelcast.jet.impl.util.ExceptionUtil.rethrow(ExceptionUtil.java:145)
        at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:533)
        at com.hazelcast.jet.impl.pipeline.transform.StreamSourceTransform.readObject(StreamSourceTransform.java:147)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at java.io.ObjectStreamClass.invokeReadObject(ObjectStreamClass.java:1184)
        at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2296)
        at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2187)
        at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1667)
        at java.io.ObjectInputStream.readObject(ObjectInputStream.java:503)
        at java.io.ObjectInputStream.readObject(ObjectInputStream.java:461)
        at java.util.HashMap.readObject(HashMap.java:1416)
        at sun.reflect.GeneratedMethodAccessor47.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at java.io.ObjectStreamClass.invokeReadObject(ObjectStreamClass.java:1184)
        at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2296)
        at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2187)
        at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1667)
        at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2405)
        at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2329)
        at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2187)
        at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1667)
        at java.io.ObjectInputStream.readObject(ObjectInputStream.java:503)
        at java.io.ObjectInputStream.readObject(ObjectInputStream.java:461)
        at com.hazelcast.internal.serialization.impl.defaultserializers.JavaDefaultSerializers$JavaSerializer.read(JavaDefaultSerializers.java:92)
        at com.hazelcast.internal.serialization.impl.defaultserializers.JavaDefaultSerializers$JavaSerializer.read(JavaDefaultSerializers.java:85)
        at com.hazelcast.internal.serialization.impl.StreamSerializerAdapter.read(StreamSerializerAdapter.java:44)
        at com.hazelcast.internal.serialization.impl.AbstractSerializationService.toObject(AbstractSerializationService.java:270)
        ... 12 more
Caused by: java.io.IOException: unexpected exception type
        at java.io.ObjectStreamClass.throwMiscException(ObjectStreamClass.java:1750)
        at java.io.ObjectStreamClass.invokeReadResolve(ObjectStreamClass.java:1280)
        at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2196)
        at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1667)
        at java.io.ObjectInputStream.readArray(ObjectInputStream.java:2093)
        at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1655)
        at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2405)
        at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2329)
        at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2187)
        at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1667)
        at java.io.ObjectInputStream.readObject(ObjectInputStream.java:503)
        at java.io.ObjectInputStream.readObject(ObjectInputStream.java:461)
        at com.hazelcast.jet.impl.pipeline.transform.StreamSourceTransform.lambda$readObject$0(StreamSourceTransform.java:149)
        at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:531)
        ... 41 more
Caused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at java.lang.invoke.SerializedLambda.readResolve(SerializedLambda.java:230)
        at sun.reflect.GeneratedMethodAccessor25.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at java.io.ObjectStreamClass.invokeReadResolve(ObjectStreamClass.java:1274)
        ... 53 more
Caused by: java.lang.IllegalArgumentException: Invalid lambda deserialization
        at com.hazelcast.jet.kinesis.KinesisSources$Builder.$deserializeLambda$(KinesisSources.java:145)
        ... 62 more
```

--- a/extensions/kinesis/src/main/java/com/hazelcast/jet/kinesis/KinesisSources.java
+++ b/extensions/kinesis/src/main/java/com/hazelcast/jet/kinesis/KinesisSources.java
@@ -327,6 +327,9 @@ public final class KinesisSources {
     }
 
     private static class DefaultProjection<T> implements BiFunctionEx<Record, Shard, T> {
+
+        private static final long serialVersionUID = 1L;
+
         @Override
         public T applyEx(Record record, Shard shard) {
             return (T) entry(record.getPartitionKey(), toArray(record));

--- a/extensions/kinesis/src/main/java/com/hazelcast/jet/kinesis/KinesisSources.java
+++ b/extensions/kinesis/src/main/java/com/hazelcast/jet/kinesis/KinesisSources.java
@@ -166,7 +166,9 @@ public final class KinesisSources {
 
         private Builder(@Nonnull String stream) {
             this.stream = stream;
-            this.projectionFn = (record, shard) -> (T) entry(record.getPartitionKey(), toArray(record, shard));
+            // Don't use lambda here since serialization/deserialization of the lambda gets corrupted
+            // after relocating aws classes
+            this.projectionFn = new DefaultProjection<>();
         }
 
         /**
@@ -322,8 +324,15 @@ public final class KinesisSources {
                     eventTimePolicy -> new KinesisSourcePMetaSupplier<T>(awsConfig, stream, retryStrategy,
                             initialShardIterators, eventTimePolicy, projectionFn));
         }
+    }
 
-        private static byte[] toArray(Record record, Shard shard) {
+    private static class DefaultProjection<T> implements BiFunctionEx<Record, Shard, T> {
+        @Override
+        public T applyEx(Record record, Shard shard) {
+            return (T) entry(record.getPartitionKey(), toArray(record));
+        }
+
+        private static byte[] toArray(Record record) {
             ByteBuffer buffer = record.getData();
             int position = buffer.position();
             int limit = buffer.limit();
@@ -334,5 +343,4 @@ public final class KinesisSources {
             }
         }
     }
-
 }

